fix warnings in pioc_support.c

There are some warnings that should be fixed:

/home/ed/pio3/ParallelIO/src/clib/pioc_support.c: In function ‘pio_get_env’:
/home/ed/pio3/ParallelIO/src/clib/pioc_support.c:195:26: warning: passing argument 2 of ‘strchr’ makes integer from pointer without a cast [-Wint-conversion]
         if(strchr(envptr,"M") != NULL){
                          ^~~
In file included from /home/ed/pio3/ParallelIO/src/clib/pio.h:15:0,
                 from /home/ed/pio3/ParallelIO/src/clib/pioc_support.c:9:
/usr/include/string.h:231:14: note: expected ‘int’ but argument is of type ‘char *’
 extern char *strchr (const char *__s, int __c)
              ^~~~~~
/home/ed/pio3/ParallelIO/src/clib/pioc_support.c:197:32: warning: passing argument 2 of ‘strchr’ makes integer from pointer without a cast [-Wint-conversion]
         }else if(strchr(envptr,"K") != NULL){
                                ^~~
In file included from /home/ed/pio3/ParallelIO/src/clib/pio.h:15:0,
                 from /home/ed/pio3/ParallelIO/src/clib/pioc_support.c:9:
/usr/include/string.h:231:14: note: expected ‘int’ but argument is of type ‘char *’
 extern char *strchr (const char *__s, int __c)
              ^~~~~~
/home/ed/pio3/ParallelIO/src/clib/pioc_support.c: In function ‘PIOc_writemap_from_f90’:
/home/ed/pio3/ParallelIO/src/clib/pioc_support.c:649:54: warning: passing argument 5 of ‘PIOc_writemap’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
     return(PIOc_writemap(file, ndims, gdims, maplen, map, MPI_Comm_f2c(f90_comm)));
                                                      ^~~
/home/ed/pio3/ParallelIO/src/clib/pioc_support.c:581:5: note: expected ‘MPI_Offset \* {aka long long int *}’ but argument is of type ‘const long long int *’
 int PIOc_writemap(const char file[], const int ndims, const int gdims[], PIO_Offset maplen, PIO_Offset map[], const MPI_Comm comm)
     ^~~~~~~~~~~~~

--- a/src/clib/pioc.c
+++ b/src/clib/pioc.c
@@ -528,9 +528,6 @@ int PIOc_Init_Intracomm(const MPI_Comm comp_comm, const int num_iotasks,
         /* Add this iosys struct to the list in the PIO library. */
         *iosysidp = pio_add_to_iosystem_list(iosys);
 
-        /* Get some info from the environment. */
-        pio_get_env();
-
         /* Allocate buffer space for compute nodes. */
         compute_buffer_init(*iosys);
 

--- a/src/clib/pioc_support.c
+++ b/src/clib/pioc_support.c
@@ -177,51 +177,6 @@ void pio_log(int severity, const char *fmt, ...)
 static pio_swapm_defaults swapm_defaults;
 bool PIO_Save_Decomps=false;
 
-/** Get PIO environment variables.
- */
-void pio_get_env(void)
-{
-    char *envptr;
-    extern bufsize pio_cnbuffer_limit;
-    envptr = getenv("PIO_Save_Decomps");
-
-    if(envptr != NULL && (strcmp(envptr,"true")==0)){
-        PIO_Save_Decomps=true;
-    }
-    swapm_defaults.nreqs = 0;
-    swapm_defaults.handshake=false;
-    swapm_defaults.isend=false;
-
-    envptr = getenv("PIO_SWAPM");
-    if(envptr != NULL){
-        char *token = strtok(envptr, ":");
-
-        swapm_defaults.nreqs = atoi(token);
-
-        token = strtok(NULL, ":");
-
-        if((token!=NULL) && strcmp(token,"t")==0){
-            swapm_defaults.handshake = true;
-        }
-        token = strtok(NULL, ":");
-
-        if((token!=NULL) && strcmp(token,"t")==0){
-            swapm_defaults.isend = true;
-        }
-        //printf("nreqs %d handshake %d isend %d\n",swapm_defaults.nreqs, swapm_defaults.handshake, swapm_defaults.isend);
-    }
-    envptr = getenv("PIO_CNBUFFER_LIMIT");
-    if(envptr != NULL){
-        int mult=1;
-        if(strchr(envptr,"M") != NULL){
-            mult = 1000000;
-        }else if(strchr(envptr,"K") != NULL){
-            mult = 1000;
-        }
-        pio_cnbuffer_limit = (bufsize) atoll(envptr) * mult;
-    }
-}
-
 /* Obtain a backtrace and print it to stderr. */
 void print_trace (FILE *fp)
 {

